lambda-runtime/src/AWS/Lambda/Runtime.hs

```diff
@@ -4,14 +4,15 @@ module AWS.Lambda.Runtime
 where
 
 import AWS.Lambda.Runtime.Prelude
+import AWS.Lambda.Runtime.Types
 import Control.Monad.Except (runExceptT)
 
 import qualified AWS.Lambda.Runtime.Client as Client
 import qualified Data.Aeson                as JSON
 
 run
   :: forall m . (MonadCatch m, MonadIO m)
-  => (JSON.Value -> m JSON.Value)
+  => (LambdaEvent -> m JSON.Value)
```

Comment:
The `requestId` in `LambdaEvent` can also be retrieved from request headers after parsing `JSON.Value`. Not, that I am opposing exposing `LambdaEvent`

Reply:
you mean response headers. But: at this point its already parsed so why not expose it.